test/TypeDecoder/nominal_types.swift

```diff
@@ -80,6 +138,38 @@ func generic<T>(_: Constrained<T>) {}
 // DEMANGLE-TYPE: $s13nominal_types11ConstrainedVyxGD
 // CHECK-TYPE: Constrained<τ_0_0>
 
+// DEMANGLE-TYPE: $s13nominal_types12STSContainerO8SubclassCyAA8STSOuterV_SiG
+// CHECK-TYPE: STSContainer<STSOuter>.Subclass<Int>
+// DEMANGLE-TYPE: $s13nominal_types0017STSContainer_swCgOA2A8STSOuterVRszrlE8SubclassCyAE_SiG
+// CHECK-TYPE: STSContainer℠<STSOuter>.Subclass<Int>
+
+// DEMANGLE-TYPE: $s13nominal_types12STSContainerO9Subclass2CyAA8STSOuterV_SiG
+// CHECK-TYPE: STSContainer<STSOuter>.Subclass2<Int>
+// DEMANGLE-TYPE: $s13nominal_types0017STSContainer_swCgOA2A8STSOuterVRszrlE9Subclass2CyAE_SiG
+// CHECK-TYPE: STSContainer℠<STSOuter>.Subclass2<Int>
+
+// DEMANGLE-TYPE: $s13nominal_types12STSContainerO9Subclass3CyAA8STSOuterV_SiG
+// CHECK-TYPE: STSContainer<STSOuter>.Subclass3<Int>
```

Comment:
Oops. This one has a constraint on it and I'm not passing a valid argument. It's working by coincidence because I wrote the wrong mangled name.